lib/immobilienscout/api/report.rb

```diff
@@ -30,6 +26,31 @@ def query_params(date_from, date_to)
         def retrieve_url(is24_id)
           "#{Immobilienscout::Client.api_url}/restapi/api/offer/v1.0/user/me/realestate/#{is24_id}/dailyreport"
         end
+
+        def fix(parsed_response)
+          response_array_fixed = place_report_daily_data_in_array(parsed_response)
+          convert_numers_from_string_to_integers(response_array_fixed)
+        end
+
+        def place_report_daily_data_in_array(parsed_response)
```

Comment:
I would name it `wrap_...` instead of `place`. This also led me to remembering that in Rails there's a helper function `Array.wrap()` which works similarly to `Array()` but properly handles Hashes etc. This way it can replace most of this function:

```
def wrap_report_daily_data_in_array(response)
  response.messages['dailyReports']['reportDailyData'] =
    Array.wrap(response.messages['dailyReports']['reportDailyData'])
end
```